lib/src/source/hosted.dart

```diff
@@ -537,6 +547,183 @@ class HostedSource extends CachedSource {
     return result;
   }
 
+  Future<List<Advisory>?> _fetchAdvisories(
+    PackageRef ref,
+    SystemCache cache,
+  ) async {
+    final description = ref.description;
+    if (description is! HostedDescription) {
+      throw ArgumentError('Wrong source');
+    }
+
+    final packageName = description.packageName;
+    final hostedUrl = description.url;
+
+    final url = _listAdvisoriesUrl(ref);
+    log.io('Get security advisories from $url.');
+
+    final String bodyText;
+    final Map<String, dynamic> body;
+    final List<Advisory>? result;
+    try {
+      bodyText = await withAuthenticatedClient(cache, Uri.parse(hostedUrl),
+          (client) async {
+        return await retryForHttp(
+            'fetching advisories for "$packageName" from "$url"', () async {
+          final request = http.Request('GET', url);
+          request.attachPubApiHeaders();
+          request.attachMetadataHeaders();
+          final response = await client.fetch(request);
+          return response.body;
+        });
+      });
+      final decoded = jsonDecode(bodyText);
+      if (decoded is! Map<String, dynamic>) {
+        throw FormatException('security advisories must be a mapping');
+      }
+      body = decoded;
+      result = _parseAdvisories(decoded, ref.name);
+    } on FormatException catch (error, stackTrace) {
+      _throwFriendlyError(error, stackTrace, packageName, hostedUrl);
+    }
+
+    // Cache the response on disk.
+    // Don't cache overly big responses.
+    if (bodyText.length < 5000 * 1024) {
+      await _cacheAdvisoriesResponse(body, ref, cache);
+    }
+    return result;
+  }
+
+  List<Advisory>? _parseAdvisories(Map body, String packageName) {
```

Comment:
Dartdoc?

Explain the meaning of the `packageName` parameter

Reply:
done.